tests/e2e/vsphere_file_volume_basic_mount.go

```diff
@@ -144,6 +144,164 @@ var _ = ginkgo.Describe("[csi-file-vanilla] Verify Two Pods can read write files
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		//cleanup for Pod1
+		defer func() {
+			if !isDeletePodAfterFileCreation {
+				ginkgo.By(fmt.Sprintf("Deleting the pod : %s in namespace %s", pod1.Name, namespace))
+				err = fpod.DeletePodWithWait(client, pod1)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				ginkgo.By("Verify volume is detached from the node")
+				isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, volHandle, pod1.Spec.NodeName)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", volHandle, pod1.Spec.NodeName))
+			}
+		}()
+
+		//Create file1.txt on Pod1
+		ginkgo.By("Create file1.txt on Pod1")
+		err = framework.CreateEmptyFileOnPod(namespace, pod1.Name, filePath1)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		//Write data on file1.txt on Pod1
+		data := "This file file1 is written by Pod1"
+		ginkgo.By("Write on file1.txt from Pod1")
+		writeDataOnFileFromPod(namespace, pod1.Name, filePath1, data)
+
+		//Delete Pod if needed
+		if isDeletePodAfterFileCreation {
+			ginkgo.By(fmt.Sprintf("Deleting the pod : %s in namespace %s", pod1.Name, namespace))
+			err = fpod.DeletePodWithWait(client, pod1)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By("Verify volume is detached from the node")
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, volHandle, pod1.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", volHandle, pod1.Spec.NodeName))
+		}
+
+		// Create Pod 2 on statically provisioned pvc
+		if staticProvisionedPVCForSecondPod {
+			// Creating label for PV.
+			// PVC will use this label as Selector to find PV
+			staticPVLabels := make(map[string]string)
+			staticPVLabels["volumeId"] = "NewVolume"
+			pv := getPersistentVolumeSpecFromVolume(volHandle, v1.PersistentVolumeReclaimDelete, staticPVLabels, v1.ReadOnlyMany)
+			pv, err = client.CoreV1().PersistentVolumes().Create(ctx, pv, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvclaim = getPersistentVolumeClaimSpecFromVolume(namespace, pv.Name, staticPVLabels, v1.ReadOnlyMany)
+			pvclaim, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(ctx, pvclaim, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		//Create Pod2 using the same pvc
+		ginkgo.By("Create Pod2 with pvc created above")
+		userid := int64(1000)
+		var pod2 *v1.Pod
+		if secondPodForNonRootUser {
+			pod2, err = CreatePodByUserID(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "", userid)
+		} else {
+			pod2, err = fpod.CreatePod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "")
```

Comment:
where do you set false for `secondPodForNonRootUser`? you have set hard value true to variable `secondPodForNonRootUser`.